# Generate .csv tables of Calibration steps

In [14]:
#make sure s = 1 in './data/datamapping.py' file 
!python3 main.py

* * datamapping module loaded!
* * a_utils module loaded!
* * b_data_transform module loaded!
* * c_pl_pw module loaded!
* * d_del_del module loaded!
* * e_error_estimation module loaded!
* * f_star_wise module loaded!
* * g_result module loaded!
* * main_modules module loaded!
* * datamapping module loaded!
* * * * * * * * * * * * * * * * * * * * 
* * * * * * * * * * * * * * * * * * * * 


Master Thesis Project: 	Galactic BVIJHK Leavitt Law Calibration for R_v = 3.23 

  			To Refine systematic errors in luminosity, distance and reddening of individual Cepheid. 
			______________________________________________________

     			Author: Shubham Mamgain (mshubham00@gmail.com) 
     			Supervisor I: Dr. Jesper Storm (AIP Potsdam)
     			Supervisor II: Prof. Dr. Maria Rosa Cioni




			To begin the calibration process, store cleaned data at ./data/input/<file_name>.csv 

			For datafile selection (k) and columns mapping, edit ./data/datamapping.py file.

			Autosave (s = 1) | Autoprocess

In [23]:
import pandas as pd


files = {}
files['0'] = '59_madore_3.23'
files['1'] = '94_jesper_3.23'
files['2'] = '20_cluster_cruz_3.23'
files['3'] = '29_LMC_3.4'
files['4'] = '32_SMC_VIJK_2.53'
files['5'] = '103_raw_data_IRSB_3.23'

k=3
r = pd.read_csv(f'data/processed/{files[str(k)]}/2_PLPW/{files[str(k)][:2]}_5_regression.csv')


In [3]:
from data.datamapping import R, R123, dis_flag, mag, colors, A
print(R, '\n', A)
for d,dis in enumerate(dis_flag):
    for a,c1 in enumerate(mag):
        for b,c2 in enumerate(mag[a+1:]):
            x = 1
            print([f'{x+c1+c2}: {R123(x,c1,c2) :.3f}' for x in mag])

* * datamapping module loaded!
{'B': 4.061402, 'V': 3.23, 'I': 1.9670699999999999, 'J': 0.9583410000000001, 'H': 0.586568, 'K': 0.397613} 
 {'B': 1.2574, 'V': 1.0, 'R': 0.845, 'I': 0.609, 'J': 0.2967, 'H': 0.1816, 'K': 0.1231}
['BBV: 4.887', 'VBV: 3.887', 'IBV: 2.367', 'JBV: 1.153', 'HBV: 0.706', 'KBV: 0.478']
['BBI: 1.940', 'VBI: 1.543', 'IBI: 0.939', 'JBI: 0.458', 'HBI: 0.280', 'KBI: 0.190']
['BBJ: 1.309', 'VBJ: 1.041', 'IBJ: 0.634', 'JBJ: 0.309', 'HBJ: 0.189', 'KBJ: 0.128']
['BBH: 1.169', 'VBH: 0.930', 'IBH: 0.566', 'JBH: 0.276', 'HBH: 0.169', 'KBH: 0.114']
['BBK: 1.109', 'VBK: 0.882', 'IBK: 0.537', 'JBK: 0.262', 'HBK: 0.160', 'KBK: 0.109']
['BVI: 3.216', 'VVI: 2.557', 'IVI: 1.557', 'JVI: 0.759', 'HVI: 0.464', 'KVI: 0.315']
['BVJ: 1.788', 'VVJ: 1.422', 'IVJ: 0.866', 'JVJ: 0.422', 'HVJ: 0.258', 'KVJ: 0.175']
['BVH: 1.537', 'VVH: 1.222', 'IVH: 0.744', 'JVH: 0.363', 'HVH: 0.222', 'KVH: 0.150']
['BVK: 1.434', 'VVK: 1.141', 'IVK: 0.695', 'JVK: 0.338', 'HVK: 0.207', 'KVK: 0.140']
['BIJ: 4

In [21]:
for a,c1 in enumerate(mag):
    for b,c2 in enumerate(mag[a+1:]):
        x=1
        print(f'{mag[0]+c1+c2} & {R123(mag[0],c1,c2)} & {mag[1]+c1+c2} & {R123(mag[1],c1,c2)} & {mag[2]+c1+c2} & {R123(mag[2],c1,c2)} & {mag[3]+c1+c2} & {R123(mag[3],c1,c2)} & {mag[4]+c1+c2} & {R123(mag[4],c1,c2)} & {mag[5]+c1+c2} & {R123(mag[5],c1,c2)} \\\\') 
        #print([f'{x+c1+c2}: {R123(x,c1,c2) :.3f}' for x in mag])

BBV & 4.887 & VBV & 3.887 & IBV & 2.367 & JBV & 1.153 & HBV & 0.706 & KBV & 0.478 \\
BBI & 1.94 & VBI & 1.543 & IBI & 0.939 & JBI & 0.458 & HBI & 0.28 & KBI & 0.19 \\
BBJ & 1.309 & VBJ & 1.041 & IBJ & 0.634 & JBJ & 0.309 & HBJ & 0.189 & KBJ & 0.128 \\
BBH & 1.169 & VBH & 0.93 & IBH & 0.566 & JBH & 0.276 & HBH & 0.169 & KBH & 0.114 \\
BBK & 1.108 & VBK & 0.882 & IBK & 0.537 & JBK & 0.262 & HBK & 0.16 & KBK & 0.108 \\
BVI & 3.216 & VVI & 2.557 & IVI & 1.557 & JVI & 0.759 & HVI & 0.464 & KVI & 0.315 \\
BVJ & 1.788 & VVJ & 1.422 & IVJ & 0.866 & JVJ & 0.422 & HVJ & 0.258 & KVJ & 0.175 \\
BVH & 1.537 & VVH & 1.222 & IVH & 0.744 & JVH & 0.363 & HVH & 0.222 & KVH & 0.15 \\
BVK & 1.434 & VVK & 1.14 & IVK & 0.694 & JVK & 0.338 & HVK & 0.207 & KVK & 0.14 \\
BIJ & 4.025 & VIJ & 3.201 & IIJ & 1.95 & JIJ & 0.95 & HIJ & 0.581 & KIJ & 0.394 \\
BIH & 2.943 & VIH & 2.341 & IIH & 1.425 & JIH & 0.694 & HIH & 0.425 & KIH & 0.288 \\
BIK & 2.587 & VIK & 2.057 & IIK & 1.253 & JIK & 0.61 & HIK & 0.374 & KIK & 

In [26]:
mag = ['B', 'V', 'I','J','H','K']
mag = ['V', 'I','J','K']
for m in mag:
    sl = r[f'{m}0'].iloc[0]
    serr = r[f'{m}0'].iloc[2]
    intr = r[f'{m}0'].iloc[1]
    ierr = r[f'{m}0'].iloc[3]
    #print(f'{m} & {sl:.2f} ({serr:.2f}) & {intr:.2f} ({ierr:.2f})\\\\')
    print(f'{m} = ({sl:.3f} \pm {serr:.3f}) (\log P -1) + {intr:.3f} \pm {ierr:.3f} \\\\')

'''
slp = r[[f'{m}0' for m in mag]].iloc[0]
print(f'Slope & {slp[0]:.2f} ({esl[0]:.3f}) & {slp[1]:.2f} ({esl[1]:.3f}) & {slp[2]:.2f} ({esl[2]:.3f}) & {slp[3]:.2f} ({esl[3]:.3f}) & {slp[4]:.2f} ({esl[4]:.3f}) & {slp[5]:.2f} ({esl[5]:.3f}) \\\\')
print(f'Intercept & {inr[0]:.2f} ({ein[0]:.3f}) & {inr[1]:.2f} ({ein[1]:.3f}) & {inr[2]:.2f} ({ein[2]:.3f}) & {inr[3]:.2f} ({ein[3]:.3f}) & {inr[4]:.2f} ({ein[4]:.3f}) & {inr[5]:.2f} ({ein[5]:.3f}) \\\\')
# wesenheit PW

cols = colors(mag)
for col in cols:#['BK', 'BI', 'VI', 'JK']:
    slp = r[[f'{m}{col}' for m in mag]].iloc[0]
    inr = r[[f'{m}{col}' for m in mag]].iloc[1]
    esl = r[[f'{m}{col}' for m in mag]].iloc[2]
    ein = r[[f'{m}{col}' for m in mag]].iloc[3]
    #print(f'{col} & {slp[0]:.2f} ({esl[0]:.3f}) & {slp[1]:.2f} ({esl[1]:.3f}) & {slp[2]:.2f} ({esl[2]:.3f}) & {slp[3]:.2f} ({esl[3]:.3f}) & {slp[4]:.2f} ({esl[4]:.3f}) & {slp[5]:.2f} ({esl[5]:.3f}) \\\\')
    #print(f'{col} & {inr[0]:.2f} ({ein[0]:.3f}) & {inr[1]:.2f} ({ein[1]:.3f}) & {inr[2]:.2f} ({ein[2]:.3f}) & {inr[3]:.2f} ({ein[3]:.3f}) & {inr[4]:.2f} ({ein[4]:.3f}) & {inr[5]:.2f} ({ein[5]:.3f}) \\\\')

for col in ['BK', 'VJ']:
    for m in ['B', 'V', 'I','J','H','K']:
        sl = r[1][f'{m}0{col}{f}'].iloc[0]
        serr = r[1][f'{m}0{col}{f}'].iloc[2]
        intr = r[1][f'{m}0{col}{f}'].iloc[1]
        ierr = r[1][f'{m}0{col}{f}'].iloc[3]
        print(f'M_{m} & = {sl:.2f}(\log P - 1)(\pm {serr:.3f})  {intr:.2f}(\pm {ierr:.3f})\\\\')
'''

V = (-2.707 \pm 0.101) (\log P -1) + 14.296 \pm 0.041 \\
I = (-2.960 \pm 0.076) (\log P -1) + 13.605 \pm 0.030 \\
J = (-3.124 \pm 0.056) (\log P -1) + 13.216 \pm 0.022 \\
K = (-3.226 \pm 0.043) (\log P -1) + 12.783 \pm 0.017 \\


"\nslp = r[[f'{m}0' for m in mag]].iloc[0]\nprint(f'Slope & {slp[0]:.2f} ({esl[0]:.3f}) & {slp[1]:.2f} ({esl[1]:.3f}) & {slp[2]:.2f} ({esl[2]:.3f}) & {slp[3]:.2f} ({esl[3]:.3f}) & {slp[4]:.2f} ({esl[4]:.3f}) & {slp[5]:.2f} ({esl[5]:.3f}) \\\\')\nprint(f'Intercept & {inr[0]:.2f} ({ein[0]:.3f}) & {inr[1]:.2f} ({ein[1]:.3f}) & {inr[2]:.2f} ({ein[2]:.3f}) & {inr[3]:.2f} ({ein[3]:.3f}) & {inr[4]:.2f} ({ein[4]:.3f}) & {inr[5]:.2f} ({ein[5]:.3f}) \\\\')\n# wesenheit PW\n\ncols = colors(mag)\nfor col in cols:#['BK', 'BI', 'VI', 'JK']:\n    slp = r[[f'{m}{col}' for m in mag]].iloc[0]\n    inr = r[[f'{m}{col}' for m in mag]].iloc[1]\n    esl = r[[f'{m}{col}' for m in mag]].iloc[2]\n    ein = r[[f'{m}{col}' for m in mag]].iloc[3]\n    #print(f'{col} & {slp[0]:.2f} ({esl[0]:.3f}) & {slp[1]:.2f} ({esl[1]:.3f}) & {slp[2]:.2f} ({esl[2]:.3f}) & {slp[3]:.2f} ({esl[3]:.3f}) & {slp[4]:.2f} ({esl[4]:.3f}) & {slp[5]:.2f} ({esl[5]:.3f}) \\\\')\n    #print(f'{col} & {inr[0]:.2f} ({ein[0]:.3f}) & {inr[1]:.2f}

In [27]:
slp = r[[f'{m}0' for m in mag]].iloc[0]
inr = r[[f'{m}0' for m in mag]].iloc[1]
esl = r[[f'{m}0' for m in mag]].iloc[2]
ein = r[[f'{m}0' for m in mag]].iloc[3]
print('Raw Leavitt Law')
for i,m in enumerate(mag):
    print(f'M_{m} & = {slp[i]:.3f}(log(P) -1)({esl[i]:.3f}) {inr[i]:.3f}\\\\')
    

Raw Leavitt Law
M_B & = -2.269(log(P) -1)(0.099) -3.214\\
M_V & = -2.663(log(P) -1)(0.082) -3.944\\
M_R & = -2.869(log(P) -1)(0.078) -4.396\\
M_I & = -2.978(log(P) -1)(0.074) -4.706\\
M_J & = -3.195(log(P) -1)(0.068) -5.258\\
M_H & = -3.330(log(P) -1)(0.073) -5.558\\
M_K & = -3.374(log(P) -1)(0.073) -5.659\\


/var/folders/_q/wljn8n6x39qc0pl27026kbvw0000gp/T/ipykernel_22581/3858159732.py:7: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print(f'M_{m} & = {slp[i]:.3f}(log(P) -1)({esl[i]:.3f}) {inr[i]:.3f}\\\\')


$$
\begin{array}{|cc|}
\hline
M_B & = -1.86(\log P - 1)(\pm 0.011)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.003)  -3.95(\pm 0.001)\\
M_I & = -2.57(\log P - 1)(\pm 0.014)  -4.74(\pm 0.004)\\
M_J & = -2.79(\log P - 1)(\pm 0.011)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.011)  -5.60(\pm 0.003)\\
M_K & = -2.97(\log P - 1)(\pm 0.011)  -5.65(\pm 0.003)\\
\hline
M_B & = -1.85(\log P - 1)(\pm 0.009)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.010)  -3.95(\pm 0.003)\\
M_I & = -2.56(\log P - 1)(\pm 0.018)  -4.73(\pm 0.005)\\
M_J & = -2.78(\log P - 1)(\pm 0.010)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.014)  -5.60(\pm 0.004)\\
M_K & = -2.97(\log P - 1)(\pm 0.014)  -5.65(\pm 0.004)\\
\hline
M_B & = -2.269(log(P) -1)(0.099) -3.214\\
M_V & = -2.663(log(P) -1)(0.082) -3.944\\
M_R & = -2.869(log(P) -1)(0.078) -4.396\\
M_I & = -2.978(log(P) -1)(0.074) -4.706\\
M_J & = -3.195(log(P) -1)(0.068) -5.258\\
M_H & = -3.330(log(P) -1)(0.073) -5.558\\
M_K & = -3.374(log(P) -1)(0.073) -5.659\\
\end{array}
$$

In [20]:
import pandas as pd
from data.datamapping import mag, col_lin, colors
files = {}
files['0'] = '59_madore_3.23'
files['1'] = '94_jesper_3.23'
files['2'] = '20_cluster_cruz_3.23'
files['3'] = '30_LMC_3.41'
files['4'] = '32_SMC_VIJK_2.74'
files['5'] = '103_raw_data_IRSB_3.23'
k=1
reg = pd.read_csv(f'data/processed/{files[str(k)]}/3_deldel/{files[str(k)][:2]}_del_slope_intercept.csv')
cols = colors(mag)
for col in cols:
    print(f"\t {col} & {reg[f'B0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'B0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'V0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'V0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'I0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'I0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'J0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'J0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'H0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'H0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'K0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'K0{col[0]}{col}'].iloc[2] : .3f}) \\\\")
    print('\t \\rowcolor{yellow!20}')
    print(f"\t    & {reg[f'B0B{col}'].iloc[0] : .3f} ({reg[f'B0B{col}'].iloc[2] : .3f}) & {reg[f'V0V{col}'].iloc[0] : .3f} ({reg[f'V0V{col}'].iloc[2] : .3f}) & {reg[f'I0I{col}'].iloc[0] : .3f} ({reg[f'I0I{col}'].iloc[2] : .3f}) & {reg[f'J0J{col}'].iloc[0] : .3f} ({reg[f'J0J{col}'].iloc[2] : .3f}) & {reg[f'H0H{col}'].iloc[0] : .3f} ({reg[f'H0H{col}'].iloc[2] : .3f}) & {reg[f'K0K{col}'].iloc[0] : .3f} ({reg[f'K0K{col}'].iloc[2] : .3f}) \\\\")
    print('\t \\hline')
    #print(f"\t{col} & {reg[f'B0B{col}'].iloc[0] : .3f} ({reg[f'B0B{col}'].iloc[2] : .3f})")

	 BV &  0.243 ( 0.084) &  0.398 ( 0.067) &  0.471 ( 0.056) &  0.479 ( 0.050) &  0.478 ( 0.045) &  0.471 ( 0.046) \\
	 \rowcolor{yellow!20}
	    &  0.243 ( 0.084) &  0.398 ( 0.067) &  0.653 ( 0.054) &  0.870 ( 0.035) &  0.929 ( 0.024) &  0.962 ( 0.017) \\
	 \hline
	 BI &  0.652 ( 0.092) &  0.758 ( 0.066) &  0.831 ( 0.045) &  0.798 ( 0.040) &  0.773 ( 0.035) &  0.765 ( 0.036) \\
	 \rowcolor{yellow!20}
	    &  0.652 ( 0.092) &  0.683 ( 0.068) &  0.832 ( 0.045) &  0.944 ( 0.025) &  0.968 ( 0.016) &  0.982 ( 0.011) \\
	 \hline
	 BJ &  0.881 ( 0.090) &  0.945 ( 0.060) &  0.975 ( 0.041) &  0.972 ( 0.021) &  0.938 ( 0.014) &  0.933 ( 0.016) \\
	 \rowcolor{yellow!20}
	    &  0.881 ( 0.090) &  0.835 ( 0.066) &  0.920 ( 0.041) &  0.972 ( 0.021) &  0.981 ( 0.014) &  0.990 ( 0.009) \\
	 \hline
	 BH &  0.917 ( 0.096) &  0.983 ( 0.065) &  1.015 ( 0.045) &  1.017 ( 0.025) &  0.988 ( 0.014) &  0.984 ( 0.014) \\
	 \rowcolor{yellow!20}
	    &  0.917 ( 0.096) &  0.859 ( 0.070) &  0.941 ( 0.043) &  0.983 (

In [7]:
r

Unnamed: 0        B0        V0        R0        I0        J0        H0  \
0        m_h -2.269164 -2.663321 -2.868529 -2.978209 -3.194569 -3.330369   
1        c_h -3.214182 -3.944345 -4.395736 -4.706254 -5.257741 -5.558140   
2    err_m_h  0.098619  0.081933  0.078214  0.073695  0.068039  0.073285   
3    err_c_h  0.029415  0.024438  0.023328  0.021981  0.020293  0.021858   

         K0       BBV       VBV  ...       JJK       HJK       KJK       BHK  \
0 -3.374274 -3.935596 -3.935596  ... -3.497598 -3.518303 -3.497933 -3.189067   
1 -5.658550 -6.300996 -6.300996  ... -5.933738 -5.977339 -5.934333 -5.322468   
2  0.073270  0.104414  0.104414  ...  0.080639  0.083925  0.080653  0.126775   
3  0.021854  0.031143  0.031143  ...  0.024052  0.025032  0.024056  0.037813   

        VHK       RHK       IHK       JHK       HHK       KHK  
0 -3.365590 -3.461932 -3.405132 -3.399588 -3.457512 -3.457812  
1 -5.553814 -5.755712 -5.684716 -5.727631 -5.849510 -5.850020  
2  0.103104  0.090063  0.077058  0.069210  0.075097  0.075101  
3  0.030752  0.026862  0.022984  0.020643  0.022399  0.022400  

[4 rows x 155 columns]

In [ ]:
reg

In [12]:
reg[[f'{m}0{m}{col}' for m in mag]]
#for i, m in enumerate(mag):
#    print(f"\t{m} &  {reg[m+col].iloc[0] : .3f} ($\pm${reg[m+col].iloc[2] : .3f}) & {reg[m+col].iloc[1] : .3f} ($\pm${reg[m+col].iloc[3] : .3f}) \\\\")

B0BHK         V0VHK         I0IHK         J0JHK         H0HHK  \
0  4.012993e-01  5.104453e-01  7.192397e-01  9.137461e-01  9.655627e-01   
1 -1.950313e-16 -5.638454e-16 -2.271485e-16 -5.805283e-16  2.815759e-16   
2  6.454443e-02  6.559318e-02  5.684106e-02  3.503537e-02  2.316720e-02   
3  3.186029e-02  2.732393e-02  1.945144e-02  1.055896e-02  6.744882e-03   

          K0KHK  
0  9.773706e-01  
1 -8.019632e-16  
2  1.523004e-02  
3  4.434356e-03

In [59]:
PLWreg = pd.read_csv(f'data/{files[str(1)]}/2_PLW/{files[str(1)][:2]}_5_regression.csv')

col='JK'
for i, m in enumerate(mag):
    print(f"\t{m},{col} &  {reg[m+col].iloc[0] : .3f} ($\pm${reg[m+col].iloc[2] : .3f}) & {reg[m+col].iloc[1] : .3f} ($\pm${reg[m+col].iloc[3] : .3f}) \\\\")

	B,JK &  -3.269 ($\pm$ 0.089) & -6.517 ($\pm$ 0.027) \\
	V,JK &  -3.338 ($\pm$ 0.082) & -6.468 ($\pm$ 0.024) \\
	I,JK &  -3.219 ($\pm$ 0.081) & -6.265 ($\pm$ 0.024) \\
	J,JK &  -3.099 ($\pm$ 0.082) & -5.952 ($\pm$ 0.024) \\
	H,JK &  -3.113 ($\pm$ 0.081) & -6.054 ($\pm$ 0.024) \\
	K,JK &  -3.099 ($\pm$ 0.082) & -5.952 ($\pm$ 0.024) \\


In [43]:
for i, m in enumerate(mag):
    print(f"\t{m} &  {b[1][m+'0'].iloc[0] : .3f} ($\pm${b[1][m+'0'].iloc[2] : .3f}) & {b[1][m+'0'].iloc[1] : .3f} ($\pm${b[1][m+'0'].iloc[3] : .3f}) \\\\")

	B &  -1.855 ($\pm$ 0.110) & -3.221 ($\pm$ 0.033) \\
	V &  -2.259 ($\pm$ 0.098) & -3.951 ($\pm$ 0.029) \\
	I &  -2.563 ($\pm$ 0.093) & -4.735 ($\pm$ 0.028) \\
	J &  -2.784 ($\pm$ 0.088) & -5.218 ($\pm$ 0.026) \\
	H &  -2.917 ($\pm$ 0.084) & -5.598 ($\pm$ 0.025) \\
	K &  -2.971 ($\pm$ 0.084) & -5.653 ($\pm$ 0.025) \\


In [9]:
from data.datamapping import mag, col_lin
ab=0
col='VI'
f='S'
b[0][[f'{m}{ab}{col}{f}' for m in mag]].style

In [15]:
riess = pd.read_csv('data/18_riess/18_true_abs_data.csv')
riess

Unnamed: 0   name      logP    EBV        plx    M_B0_g    M_V0_g  \
0            0  CFCas  0.687994  0.553  12.252555 -2.254463 -2.910745   
1            1  CSVel  0.771201  0.737  12.547220 -2.616688 -3.218730   
2            2  CVMon  0.730685  0.722  10.978801 -2.437799 -3.005861   
3            3  EVSct  0.490098  0.655  11.225259 -2.706761 -3.196909   
4            4  GHLup  0.967448  0.347  10.222618 -2.846879 -3.706428   
5            5  QZNor  0.578244  0.263  11.425521 -2.782353 -3.406011   
6            6  RSOri  0.878917  0.352  11.001908 -3.115326 -3.724868   
7            7   SNor  0.989194  0.179   9.728963 -3.111366 -3.878133   
8            8  STTau  0.605768  0.368  10.141119 -2.632237 -3.110759   
9            9  SVVul  1.652569  0.462  11.815498 -5.070358 -6.094758   
10          10  TWNor  1.032868  1.157  11.971017 -3.236631 -4.007127   
11          11   USgr  0.828997  0.402   8.923041 -2.838023 -3.531501   
12          12   VCen  0.739882  0.294   9.197433 -2.736436 -3.313053   
13          13   VLac  0.697493  0.335  11.372475 -2.973960 -3.533525   
14          14  AAGem  1.053181  0.309  12.300635 -2.799106 -3.569705   
15          15   XLac  0.735997  0.336  11.290871 -3.406587 -3.971151   
16          16   XPup  1.414321  0.421  11.818071 -3.897448 -4.670901   
17          17   YSct  1.014575  0.757  11.143930 -3.153024 -3.957040   

      M_I0_g    M_J0_g    M_H0_g    M_K0_g  
0  -3.647558 -4.073122 -4.443855 -4.499111  
1  -3.926570 -4.402329 -4.744093 -4.812501  
2  -3.780693 -4.255762 -4.609903 -4.680316  
3  -3.888575 -4.179029 -4.437192 -4.449022  
4  -4.540071 -5.057895 -5.465485 -5.542995  
5  -4.077011 -4.537572 -4.846279 -4.907611  
6  -4.440180 -4.844901 -5.189698 -5.220207  
7  -4.653490 -5.168789 -5.558612 -5.635765  
8  -3.725812 -4.136202 -4.458263 -4.494567  
9  -7.052792 -7.583238 -7.995597 -8.063077  
10 -4.956180 -5.556253 -5.953434 -6.025733  
11 -4.286504 -4.715191 -5.065062 -5.123557  
12 -3.978802 -4.401722 -4.741315 -4.801438  
13 -4.164361 -4.567434 -4.897326 -4.914239  
14 -4.326461 -4.858071 -5.281285 -5.313405  
15 -4.633721 -4.999772 -5.330306 -5.374019  
16 -5.537848 -6.035141 -6.464200 -6.548891  
17 -4.789557 -5.279902 -5.690495 -5.743898

In [2]:
# madore result
Band & Slope  & $\sigma $ (mag)  & Zero  & $\sigma$(mag)  & rms \\
B & −2.277 & 0.121 & −3.214 & 0.036 & 0.27 //
V& −2.670 & 0.095 & −3.944  & 0.028  & 0.21 //
R& −2.874 & 0.075 & −4.396  & 0.022 &  0.17 //
I& −2.983 & 0.064 & −4.706  & 0.019  & 0.14 //
J& −3.198 &  0.054 & −5.258 &  0.016 &  0.12 //
H& −3.333 & 0.055 & −5.558  & 0.017  & 0.12 //
K& −3.377 & 0.055 & −5.659  & 0.016 &  0.12 //
W, VI & −3.476 & 0.037 & −5.889 &  0.011  & 0.08 //
W, BI & −3.600 & 0.033 & −5.997 &  0.010  & 0.07  //

$$
\begin{array}{|cc|}
\hline
M_B & = -1.86(\log P - 1)(\pm 0.011)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.003)  -3.95(\pm 0.001)\\
M_I & = -2.57(\log P - 1)(\pm 0.014)  -4.74(\pm 0.004)\\
M_J & = -2.79(\log P - 1)(\pm 0.011)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.011)  -5.60(\pm 0.003)\\
M_K & = -2.97(\log P - 1)(\pm 0.011)  -5.65(\pm 0.003)\\
\hline
M_B & = -1.85(\log P - 1)(\pm 0.009)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.010)  -3.95(\pm 0.003)\\
M_I & = -2.56(\log P - 1)(\pm 0.018)  -4.73(\pm 0.005)\\
M_J & = -2.78(\log P - 1)(\pm 0.010)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.014)  -5.60(\pm 0.004)\\
M_K & = -2.97(\log P - 1)(\pm 0.014)  -5.65(\pm 0.004)\\
\hline
\end{array}
$$



$$
\begin{array}{|cc|}
\hline
M_B & = -1.86(\log P - 1)(\pm 0.011)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.003)  -3.95(\pm 0.001)\\
M_I & = -2.57(\log P - 1)(\pm 0.014)  -4.74(\pm 0.004)\\
M_J & = -2.79(\log P - 1)(\pm 0.011)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.011)  -5.60(\pm 0.003)\\
M_K & = -2.97(\log P - 1)(\pm 0.011)  -5.65(\pm 0.003)\\
\hline
M_B & = -1.85(\log P - 1)(\pm 0.009)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.010)  -3.95(\pm 0.003)\\
M_I & = -2.56(\log P - 1)(\pm 0.018)  -4.73(\pm 0.005)\\
M_J & = -2.78(\log P - 1)(\pm 0.010)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.014)  -5.60(\pm 0.004)\\
M_K & = -2.97(\log P - 1)(\pm 0.014)  -5.65(\pm 0.004)\\
\hline
\end{array}
$$
